src/Console/Command/Hook/PrepareCommitMsg.php

```diff
@@ -54,6 +57,19 @@ protected function configure()
      */
     protected function setup(InputInterface $input, OutputInterface $output, Config $config, Git\Repository $repository)
     {
-        // TODO: the message preparation settings have to be stored in the repository object
+        $this->file = $input->getArgument('file');
+
+        $gitConfig        = $repository->getConfigOperator();
+        $commentCharacter = $gitConfig->getSafely('core.commentchar', '#');
+
+        $repository->setCommitMsg(Git\CommitMessage::createFromFile($this->file, $commentCharacter));
```

Comment:
Additionally a functionality to add arbitrary information in the repository might be a good idea. Something like `$repository->setVariable('name', 'value')` or `$repository->setOptions(new PrepareCommitMessageOptions(...)))` and `PrepareCommitMessageOptions` extends `ArrayObject` or a common interface...